### PR TITLE
feat(inference): add RLDX-1 OpenVINO preprocessing pipeline and VTC window callback

### PR DESCRIPTION
**Pull request by eugene123tw**
## Summary
This PR adds the runtime-side processing stack needed to run exported RLDX-1 policies with OpenVINO, including VTC frame-window assembly, prompt/token composition, and dynamic M-RoPE position-id generation for Qwen3-VL style inputs.

## What Changed
- Added a VTC frame-window callback for inference-time temporal history assembly.
- Added an RLDX-1 image preprocessor that:
  - builds frozen-geometry pixel values for Qwen3-VL style inputs
  - emits image grid metadata
  - normalizes task text
  - pads/coerces state inputs to the expected shape
- Added a tokenizer-contract-based token composer to inject multimodal wrapper/special token segments and produce fixed-length prompt ids + masks.
- Added a dynamic RoPE preprocessor that:
  - repads to left-padding layout
  - appends cognition placeholder tokens
  - computes 3-axis M-RoPE position ids in NumPy
  - emits model-ready input_ids, position_ids, and attention_mask
- Registered new preprocessors and callback in the inference component registry and exports.
- Added focused unit tests for:
  - RLDX-1 preprocessor behavior
  - tokenizer contract composition behavior
  - NumPy RoPE parity behavior
  - VTC callback lifecycle/reset behavior

## Why
RLDX-1 exported graphs require strict prompt/image layout consistency and runtime-generated position ids. This PR moves those requirements into explicit runtime components so exported models can run reliably across CPU/GPU OpenVINO targets.

## Example Usage
~~~python
policy = Rldx1(
        pretrained_name_or_path="RLWRLD/RLDX-1-FT-LIBERO",
        input_features={
            "front_view": Feature(ftype=FeatureType.VISUAL, shape=(3, 256, 256), name="front_view"),
            "left_wrist_view": Feature(ftype=FeatureType.VISUAL, shape=(3, 256, 256), name="left_wrist_view"),
        },
    )

    policy.export("rldx1_libero_finetuned_hf_ov", backend="openvino")
    policy.to_openvino("rldx1_libero_finetuned_hf_ov")

    model = InferenceModel(
        export_dir="./rldx1_libero_finetuned_hf_ov",
        device="CPU",  # or "GPU" for Intel iGPU/dGPU
        callbacks=[Rldx1VtcWindowCallback(video_length=4, video_stride=2)]
    )

    benchmark = LiberoBenchmark(
        task_suite="libero_10",   # --env.task=libero_10
        task_ids=[6],             # --env.task_ids=[6] — reproduces libero_10_6
        num_episodes=5,          # --env.num_episodes=10
        max_steps=500,            # match the 500 steps seen in the failing run
        video_dir="./tmp_scripts/ov_libero_eval_videos",
        record_mode="all",
    )
    results = benchmark.evaluate(model)
~~~

## Benchmark Note
Panther Lake performance snapshot:
- Avg FPS: 16.5

## Validation
- Added unit coverage for new RLDX-1 inference components and callback integration.
- Existing inference callback wiring coverage remains in place for lifecycle behavior.

## Risks / Notes
- Tokenizer contract and model export artifacts must remain in sync (special ids, prefix/suffix ids, max token length, image block assumptions).
- VTC callback assumes stable camera keys within an episode and reset calls at episode boundaries.

### CODE REVIEW

File: src/physicalai/inference/preprocessors/rldx1_rope.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            extended_ids,
+            grid_thw,
+            extended_mask,
+            image_token_id=self._image_token_id,
+            vision_start_token_id=self._vision_start_token_id,
+            spatial_merge_size=self._spatial_merge_size,
+        )
+
+        outputs = dict(inputs)
+        outputs[INPUT_IDS] = left_ids
```
[the remark this answers is not in the document]
Comment (eugene123tw) [reply src/physicalai/inference/preprocessors/rldx1_rope.py:97]: If you have doubt about this.... 

`left_ids` is correct. `extended_ids` would be wrong. Here's why:

The exported model's input_ids port is 1024-wide, but position_ids/attention_mask are 1088-wide. From the OV port:

input_ids       [1, 1024]
position_ids    [3, 1, 1088]
attention_mask  [1, 1088]

File: src/physicalai/inference/callbacks/__init__.py [hunk 2]
```diff
@@ -10,12 +10,14 @@
...
 from physicalai.inference.callbacks.latency import LatencyMonitor
+from physicalai.inference.callbacks.rldx1_vtc import Rldx1VtcWindowCallback
 from physicalai.inference.callbacks.rtc_latency import RTCLatencyTracker
 from physicalai.inference.callbacks.throughput import ThroughputMonitor
 
 __all__ = [
     "Callback",
     "LatencyMonitor",
     "RTCLatencyTracker",
+    "Rldx1VtcWindowCallback",
```
Comment (samet-akcay) [inline src/physicalai/inference/callbacks/__init__.py:21]: based on this example 

```python
    model = InferenceModel(
        export_dir="./rldx1_libero_finetuned_hf_ov",
        device="CPU",  # or "GPU" for Intel iGPU/dGPU
        callbacks=[Rldx1VtcWindowCallback(video_length=4, video_stride=2)]
    )
```

what happens if a user forgets to add the callback? 

is this callback added to the `manifest.json` file as well? say a user would just use manifest file to build the inference pipeline ?

Comment (eugene123tw) [reply src/physicalai/inference/callbacks/__init__.py:21]: @samet-akcay Today, callback wiring is a runtime constructor concern, not part of the manifest-driven inference pipeline.

What happens if a user forgets to add `Rldx1VtcWindowCallback`:

- If they forgot the callback, it might produce incorrect temporal behavior because the model expects multi-frame context.

Is this callback stored in manifest.json:
- Not currently. The manifest supports runner plus sequential preprocessors and postprocessors, but not inference callbacks as a declared pipeline stage.

Current manifest spec: 

```py
class ModelSpec(BaseModel):
    """Model inference specification.

    Groups the runner, artifacts, preprocessors, and postprocessors
    under a single ``model`` section in the manifest.

    Attributes:
        n_obs_steps: Number of observation steps the model expects.
        runner: Runner component specification.
        artifacts: Mapping of logical name → filename,
            e.g. ``{"model": "model.onnx"}``.
        preprocessors: Pipeline stages applied to observations
            *before* the runner.
        postprocessors: Pipeline stages applied to runner output
            *after* inference.
        input_features: Declared input feature descriptors.
        output_features: Declared output feature descriptors.
    """

    model_config = ConfigDict(frozen=True)
    n_obs_steps: int = 1
    runner: ComponentSpec | None = None
    artifacts: dict[str, str] = Field(default_factory=dict)
    preprocessors: list[ComponentSpec] = Field(default_factory=list)
    postprocessors: list[ComponentSpec] = Field(default_factory=list)
    input_features: list[ComponentSpec] = Field(default_factory=list)
    output_features: list[ComponentSpec] = Field(default_factory=list)
```

- So a user who only loads the manifest gets pre/post processing from manifest, but does not automatically get this callback. This is the current limitation of manifest.

I suggest:
- Keep this PR scoped as-is.
- Add manifest-level callback support in a follow-up PR, then we can revisit and make `Rldx1VtcWindowCallback` declarative there.

Comment (samet-akcay) [reply src/physicalai/inference/callbacks/__init__.py:21]: Created the issue in #266

**Review by samet-akcay — APPROVED** [review]

**Comment by MarkRedeman** [comment]
@eugene123tw regarding Samet's comment on `Rldx1VtcWindowCallback` does this mean we also need to add a custom configuration in Studio's inference runtime, or was this already done?
